### PR TITLE
os/bluestore: put strings in mempool

### PR DESCRIPTION
**Pull request by liewegas**
This adds string support to mempool, and then moves a few bluestore strings
into the blustore_meta_other pool.  I don't think there are any remaining
strings of interest.

### CODE REVIEW

**Comment by liewegas** [comment]
subsumes https://github.com/ceph/ceph/pull/12290

**Comment by xiexingguo** [comment]
Need rebase

**Comment by xiexingguo** [comment]
This fails the mempool UT

```
The following tests FAILED:
	 35 - unittest_mempool (Failed)
	116 - osd-crush.sh (Failed)

**Comment by ifed01** [comment]
@liewegas  - any feedback on my previous comments in this PR?

File: src/kv/KeyValueDB.h [hunk 1]
```diff
@@ -56,7 +56,13 @@ class KeyValueDB {
       const std::string &k,	      ///< [in] Key to set
       const bufferlist &bl    ///< [in] Value to set
       ) = 0;
-
+    virtual void set(
+      const std::string &prefix,
```
Comment (liewegas) [inline src/kv/KeyValueDB.h]: Good idea.  Let's do that with a separate changeset, though?

**Comment by liewegas** [comment]
Hmm, unit test passes for me.  :/

File: src/os/kstore/KStore.cc [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 #include "KStore.h"
 #include "osd/osd_types.h"
-#include "kv.h"
+#include "os/bluestore/kv.h"
```
Comment (liewegas) [inline src/os/kstore/KStore.cc]: fixed

File: src/kv/KeyValueDB.h (hunk 1 above)
Comment (ifed01) [inline src/kv/KeyValueDB.h]: Sure.

File: src/os/kv.h [hunk 3]
```diff
@@ -1,13 +1,15 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
-#include "kv.h"
+#ifndef CEPH_OS_BLUESTORE_KV_H
```
Comment (ifed01) [inline src/os/kv.h]: CEPH_OS_BLUESTORE_KV_H -> CEPG_OS_KV_H
